force-app/main/service/DonationHistoryService.cls

```diff
@@ -0,0 +1,160 @@
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2021
+* @description Donation History Service class in NPSP.
+*/
+public with sharing class DonationHistoryService {
+
+    static final String HARD_CREDIT = 'Hard Credit';
+    static final String SOFT_CREDIT = 'Soft Credit';
+    static final String PARTIAL_SOFT_CREDIT = 'Partial Soft Credit';
+    static final String NPSP_CREDIT_ROLES= 'NPSP_OppsSoftCreditRoles';
+    
+    @TestVisible
+    private UTIL_Permissions permissions {
+        get {
+            if(permissions == null) {
+                permissions = new UTIL_Permissions();
+            }
+            return permissions;
+        }
+        set;
+    }
+
+    private Boolean hasAccessToDonationHistory(){
+        return permissions.canRead(Opportunity.SObjectType, new Set<SObjectField>{
+            Opportunity.Amount,
+            Opportunity.CloseDate,
+            Opportunity.Name
+        }) && permissions.canRead(Contact.SObjectType, new Set<SObjectField>{
+            Contact.Email,
+            Contact.Name
+        }) && permissions.canRead(Partial_Soft_Credit__c.SObjectType, new Set<SObjectField>{
+            Partial_Soft_Credit__c.Amount__c,
+            Partial_Soft_Credit__c.Contact_Role_ID__c
+        });
+    }
+
+    /** 
+    * @author Salesforce.org
+    * @date 2021
+    * @description method to return Donation History list.
+    */
+    public List<DonationHistoryController.DonationHistory> getDonationHistory(Id contactId){
+        if (hasAccessToDonationHistory()) {
+            List<OpportunityContactRole> credits = new DonationHistorySelector().getWonOpportunityContactRolesByContactId(contactId);
```

Comment:
```suggestion
            List<OpportunityContactRole> credits =
                new DonationHistorySelector().getWonOpportunityContactRolesByContactId(contactId);
```

Non-blocking but ideally break up lines longer than 120 characters for better readability.

Reply:
break up added for lines longer than 120 characters!